sklearn/svm/_classes.py

```diff
@@ -670,6 +670,15 @@ class SVC(BaseSVC):
 
         .. versionadded:: 1.0
 
+    n_iter_ : int or ndarray of shape (n_class * (n_class - 1) // 2,)
+        Number of iterations run by the optimization routine to fit the model.
+        If `classes_ <= 2`, only one model is optimized, thus an integer is
+        returned. Otherwise, multiple models are optimized separately, thus
+        having multiple number of iterations. In this case a numpy array is
+        returned with the number of iterations for each model.
```

Comment:
have we discussed weather this is a good idea? Here we're depending the type of the output to the input, which I'm not sure if it's an entirely good idea. Why not always return an ndarray? It would also simplify the implementation logic.

Reply:
Yes, I think this should be something to discuss.

I think we have three options:
1. Always return an array (or list). This approach advantage is that the code is simpler and we expose the number of iterations of all the models. The problem is that we make it a little harder for the user to interact with the attribute. For example, when training an `SVR`, the `n_iter_` will always have one element, but the user will have to get the only element of the array to get the integer. This could be confusing since for almost every other estimator with a unique `n_iter_` this is not the case. Another estimator that takes this approach is the LogisticRegression.
2. Depending on the number of models optimized, return an array or an integer. The advantage of this is that the user has direct access to the number of iterations when there is only one model and that we always expose the number of iterations of all the models. The disadvantage is that we have two attributes (one public and another private) that represent the same information, i.e., the number of iterations. Also, here we have a little more code to transform the type of the object depending on the number of models. It should be noted that this approach only adds 5 to 6 lines of code compared to the previous option. Other estimators that take this approach are the `ElasticNet` and `Lars` and the subclasses that inherit from them.
3. Always return an integer. When there is more than one model, the maximum number of iteration across all models is returned. This always returns an integer, but we lose information and we have to add more logic to the codebase (more or less the same number of lines as with the previous approach). Another estimator that takes this approach is `FastICA`.

Reply:
What do you think @adrinjalali, @ogrisel, @jjerphan?

In my case, the only option I would discard is the third one because we lose information.